src/token_classifier.rs

```diff
@@ -57,6 +57,7 @@ where
 
 fn classify_separator(c: char) -> Option<SeparatorKind> {
     match deunicode_char(c)?.chars().next()? {
+        _ if matches!(c, 'а'..='я' | 'А'..='Я') => None, // prevent deunicoding cyrillic chars (e.g. ь -> ' is incorrect)
```

Comment:
can you add separate ranges for cyrillic? According to the [unicode table](https://unicode-table.com/fr/#ipa-extensions) there are a bunch of character in this table that may be considered as separator.

Reply:
hm, actually the cyrillic part of the table is only that one
![image](https://user-images.githubusercontent.com/5773434/114731770-ad907b80-9d4a-11eb-9a93-7c66241f2442.png)
which I match by `'а'..='я' | 'А'..='Я'`.

I think that's enough and we can use previous deunicode way for detecting separators?

Reply:
uhm, so this A as a different UTF8 code as the the latin A, very confusing... Could you use hex range instead, and add a comment to tell that this correspond to the cyrillic range? This will avoid any confusion in the future :)

Reply:
haha yeah sure. Fixed.